### ISSUE TITLE
Working on unsupported datatype Or Internal compiler abort

### ISSUE THREAD

**Issue by DLMasterCat** [google-coral/edgetpu#65]
Overall, I tried to convert model from pytorch->onnx->tensorflow->tensorflow lite, as the issue I posted here.
https://github.com/tensorflow/tensorflow/issues/36610
Every tflite file works fine on interpreter but can not be mapped to TPU.
I also try to build it based on "keras model" of tensorflow==2.1.0、1.15.0, and the problem is still there.
I have no idea why this happens, even all the operations are compatible on official website.

Actually, this model is a simpler case of DenseNet. and I successfully make DenseNet map to TPU with the latest compiler version....

![擷取](https://user-images.githubusercontent.com/60849247/75334455-dd6c8d80-58c2-11ea-99fe-679fca998e9a.PNG)
![擷取2](https://user-images.githubusercontent.com/60849247/75334457-df365100-58c2-11ea-813a-a4a2e6c20f7d.PNG)

Is it possible to get older version of edgetpu_compiler?

Attach files are these tflite models, and the script how I build the keras model.

https://drive.google.com/file/d/1f3sMJwEhiisgk368RBWzxgv-HqOOAfqr/view?usp=sharing

Thanks!!!

**Comment by Namburger** [comment]
@DLMasterCat 
Could you verify whether if this is an inference model or training model? 
Are results correct after converting from onnx->tensorflow and also after tensorflow->tflite? When using the [visualized](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/tools/visualize.py) tool, I noticed that the scaling is abnormally small for `tf_op_layer_concat_103/concat_103`, which is where compiler aborted:
```{'quantized_dimension': 0, 'scale': [1.23983355e-05], 'min': [-0.0015250447], 'max': [0.001636531], 'details_type': 0, 'zero_point': [-5], 'details': None}
```
Here is the full outputs of visualized: 
[DS_39_320_quant_int8_1.15K_selfconstr.html.tar.gz](https://github.com/google-coral/edgetpu/files/4257826/DS_39_320_quant_int8_1.15K_selfconstr.html.tar.gz)

**Comment by DLMasterCat** [comment]
@Namburger 
Thanks for your reply and hint:D
Most of these models haven't been trained because I just want to test whether their architecture works.
Here is the trained model and the result of visualize tools. 
[model_1_15_trained.zip](https://github.com/google-coral/edgetpu/files/4259511/model_1_15_trained.zip)
The scale of concat operation is normal now.
Actually, it is surprise to me that scale of operation is the reason of compiler abort.

**Comment by Namburger** [comment]
@DLMasterCat You can clone this repo, check out an older commit and use [these binaries](https://github.com/google-coral/edgetpu/tree/master/compiler) to get an older compiler, unfortunately it doesn't go back to before runtime 12 and we also don't recommend rewinding the compiler version.

**Comment by DLMasterCat** [comment]
@Namburger 
Sorry for my stupid questions... I still have some problems to convert it well.
I change the commit version to "diploria2 release"(without minor fix), and the version of compiler became 2.0.267685300. However, it still aborted.
Thus, I have two questions about this situation.
First,  Would "if...else..." in the call method be implemented during the inference? Or the flow of data during inference is decided by each node. If yes, would it become a problem when compiled with edgetpu compiler?

Second, I saw some information that they didn't recommend use keras API to build model, especially in the Quantization-aware training case. Thus...if I want to build model by keras.layer, should I still use the post quantization one?

Thank you for being so kind ><

**Comment by DLMasterCat** [comment]
@Namburger 
Sorry... I still can not compile it..
Does the scale of Something like concat operation, or bias of Conv2D always make compiler abort?

And, I don't understand the limitation of input of concat.
If the input is output of previous layer, such as Batchnorm, which will not be "a compile-time constant tensor"?


Thanks!!

**Comment by Namburger** [comment]
Sorry it has been a while, but this seems like a tflite conversion issue rather than compiler. 
My suggestion, maybe try post training quantization on it again with a newer version of tensorflow. 
Closing for now
